### PR TITLE
using jdk feature makes code simple

### PR DESCRIPTION
**Pull request by LINGLUOJUN**
## What is the purpose of the change

using jdk feature makes code simple

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@songxiaosheng @wxbty PTAL

**Comment by songxiaosheng** [comment]
please fix ci problem 
```
[INFO] Starting audit...
Error:  /home/runner/work/dubbo/dubbo/dubbo/dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/DefaultMetricsCollector.java:48:8: Unused import - java.util.concurrent.atomic.AtomicLong. [UnusedImports]
Audit done.
[INFO] There is 1 error reported by Checkstyle 8.45.1 with codestyle/checkstyle.xml ruleset.
Error:  src/main/java/org/apache/dubbo/metrics/collector/DefaultMetricsCollector.java:[48,8] (imports) UnusedImports: Unused import - java.util.concurrent.atomic.AtomicLong.
```

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/MetricsGlobalRegistry.java [hunk 1]
```diff
@@ -38,12 +37,10 @@ public class MetricsGlobalRegistry {
...
      */
     public static CompositeMeterRegistry getCompositeRegistry(ApplicationModel applicationModel) {
-        Optional<MetricsConfig> configOptional = applicationModel.getApplicationConfigManager().getMetrics();
-        if (configOptional.isPresent() && configOptional.get().getUseGlobalRegistry() != null && configOptional.get().getUseGlobalRegistry()) {
-            return Metrics.globalRegistry;
-        } else {
-            return compositeRegistry;
-        }
+        return applicationModel.getApplicationConfigManager().getMetrics()
+            .filter(MetricsConfig::getUseGlobalRegistry)
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/MetricsGlobalRegistry.java:41]: useGlobalRegistry may be null, this filter will trigger NPE

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java [hunk 2]
```diff
@@ -72,13 +73,12 @@ public void addExecutors(String name, ExecutorService executorService) {
...
-        List<MetricSample> metricSamples = new ArrayList<>();
-
-        sampleThreadPoolExecutor.forEach((name, executor) -> {
-            metricSamples.addAll(createMetricsSample(name, executor));
-        });
+        return sampleThreadPoolExecutor.entrySet()
+            .stream()
+            .map(ele -> createMetricsSample(ele.getKey(), ele.getValue()))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java:81]: A map can be omitted. sampleThreadPoolExecutor.entrySet().stream()
             .flatMap(entry -> createMetricsSample(entry.getKey(), entry.getValue()).stream())
             .collect(Collectors.toList());

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java [hunk 3]
```diff
@@ -47,28 +48,23 @@ public ThreadRejectMetricsCountSampler(DefaultMetricsCollector collector) {
...
-                list.add(getGaugeMetricSample(metricsKey, k, THREAD_POOL, v, AtomicLong::get))));
+        return metricNames
+            .stream()
+            .map(metricName -> getCount(metricName)
+                .entrySet()
+                .stream()
+                .map(ele -> getGaugeMetricSample(MetricsKey.THREAD_POOL_THREAD_REJECT_COUNT, ele.getKey(), THREAD_POOL, ele.getValue(), AtomicLong::get))
+                .collect(Collectors.toList()))
+            .filter(CollectionUtils::isNotEmpty)
+            .flatMap(Collection::stream).collect(Collectors.toList());
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java:67]: This nesting is too complicated. Please define the code in the map as a method and give it a meaningful name.

**Review by wxbty — CHANGES_REQUESTED** [review]

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/MetricsGlobalRegistry.java (hunk 1 above)
Comment (LINGLUOJUN) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/MetricsGlobalRegistry.java]: already fixed

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java (hunk 2 above)
Comment (LINGLUOJUN) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java:79]: ok, and it already fixed

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java (hunk 3 above)
Comment (LINGLUOJUN) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java]: fixed

**Pull request closed without merging**